documents/coredocument.go

```diff
@@ -195,6 +215,22 @@ func (cd *CoreDocument) NextVersion() []byte {
 	return cd.Document.NextVersion
 }
 
+// GetStatus returns document status
+func (cd *CoreDocument) GetStatus() status {
+	return cd.Status
+}
+
+// SetStatus set the status of the document.
+// if the document is already committed, returns an error if set status is called.
+func (cd *CoreDocument) SetStatus(st status) error {
+	if cd.Status == Committed && st != Committed {
+		return ErrCDStatus
+	}
```

Comment:
Should we add as well a transition check on `committing`? So a state lower than that one, like `pending` cannot be set either? 
On those lines, if the committing fails, should we update the model to `pending` again? or to errored?

Reply:
> On those lines, if the committing fails, should we update the model to pending again? or to errored?
This should be handled by the caller. We cannot assume it to be Pending or Committing

Reply:
> Should we add as well a transition check on committing? So a state lower than that one, like pending cannot be set either?

I dont think this is needed ATM since the committing state will be seet by the `Commit` function of the documents Service. and Pending state is handled by the pending document service